src/unix/mod.rs

```diff
@@ -27,7 +27,7 @@ pub type uid_t = u32;
 pub type gid_t = u32;
 pub type in_addr_t = u32;
 pub type in_port_t = u16;
-pub type sighandler_t = ::size_t;
+pub type sighandler_t = __c_anonymous_sigaction_handler;
```

Comment:
Is the current type _wrong_ on all the UNIX platforms?

Reply:
I believe so. Here's what I can dig up so far:

BSD/XNU uses one of the function pointer variants `void (*sig_t)(int)`, and calls it `sig_t`:

https://github.com/apple/darwin-xnu/blob/main/bsd/sys/signal.h#L500
https://github.com/NetBSD/src/blob/trunk/sys/sys/signal.h#L172

Haiku has both function pointer variants:

https://github.com/haiku/haiku/blob/d0e006f5569f24e73a4d83f5c9766f1132160d7a/headers/posix/signal.h#L61-L62

Hermit has the `void (*signal_handler_t)(int)`:

https://github.com/hermitcore/libhermit/blob/63e67e1af1e56fbb51f14c89997bbaac11a37cfa/include/hermit/signal.h#L46

Redox as well:

https://github.com/redox-os/relibc/blob/29e2f2923170a7a52c5645ab002564bfa4b0c43b/src/header/signal/mod.rs#L35